r/R/record-batch.R

```diff
@@ -161,6 +161,17 @@ RecordBatch$create <- function(..., schema = NULL) {
     out <- RecordBatch__from_arrays(schema, arrays)
     return(dplyr::group_by(out, !!!dplyr::groups(arrays[[1]])))
   }
+
+  # If any arrays are length 1, recycle them  
+  arr_lens <- map(arrays, length)
+  if (length(arrays) > 1 && any(arr_lens == 1) && !all(arr_lens==1)){
+    max_array_len <- max(unlist(arr_lens))
```

Comment:
```suggestion
  arr_lens <- map_int(arrays, length)
  if (length(arrays) > 1 && any(arr_lens == 1) && !all(arr_lens==1)){
    max_array_len <- max(arr_lens)
```
(and the same in `table.R`)